src/main/java/seedu/address/model/person/DoB.java

```diff
@@ -0,0 +1,75 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.FormatStyle;
+
+/**
+ * Represents a Person's date of birth in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidDoB(String)}
+ */
+public class DoB {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Date of births should only contain numeric characters in the format yyyy-mm-dd";
+
+    /*
+     * The first character of the address must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
```

Comment:
Not sure what this comment is referring to, maybe we should remove it or make it more explicit?